spec/requests/v1/api_spec.rb

```diff
@@ -367,6 +367,256 @@
+
+        it "should return a HTTP 200 status code" do
+          expect(@response.status).to eq(200)
+        end
+
+        it "should return JSON representing the new task item" do
+          parsed_json = JSON.parse(@response.body)
+          expect(parsed_json["name"]).to eq(params[:task_item][:name])
+          expect(parsed_json["external_id"]).to eq(task_item.external_id)
+          expect(parsed_json["completed_at"]).to_not be_blank
+        end
+
+        it "should update the task item with specified fields" do
+          t = TaskItem.find(task_item.id)
+          expect(t.name).to eq(params[:task_item][:name])
+          expect(t.completed_at).to_not be_nil
+          expect(t).to be_completed
+        end
+
+        it "should not change fields that are not specified" do
+          task_item2 = TaskItem.find(task_item.id)
+          expect(task_item2.external_id).to eq(task_item.external_id)
+          expect(task_item2.url).to eq(task_item.url)
+        end
+      end
+
+      context "when the token is invalid" do
+        before { @response = put "/api/v1/tasks/#{task.id}/task_items/#{task_item.id}", params, empty_header }
+
+        it "should return a HTTP 401 status" do
+          expect(@response.status).to eq(401)
+        end
+
+        it "should not update the task item" do
+          task_item2 = TaskItem.find(task_item.id)
+          expect(task_item2).to eq(task_item)
+        end
+      end
+    end
+
+    describe "DELETE /api/v1/tasks/:id/task_items/:item_id" do
+      let(:task) do
+        Task.create! do |t|
+          t.name = 'New Task'
+          t.url = 'http://www.gsa.gov'
+          t.user_id = user.id
+          t.app_id = client_app.id
+          t.task_items = [
+            TaskItem.new(name: "Task Item #1", "external_id": "abcdef")
+          ]
+        end
+        let(:task_item) { task.task_items.first }
+
+        context "when the token is valid" do
+          before { @response = delete "/api/v1/tasks/#{task.id}/task_items/#{task_item.id}", nil, header }
+
+          it "should return a HTTP 200 status code" do
+            expect(@response.status).to eq(200)
+          end
+
+          it "should return a JSON representation of the task item" do
+            parsed_json = JSON.parse(@response.body)
+            expect(parsed_json["name"]).to eq(task_item["name"])
+            expect(parsed_json["external_id"]).to eq(task_item["abcdef"])
+          end
+
+          it "should delete the task_item" do
+            expect(TaskItem.where(id: task_item.id).count).to eq(0)
+          end
+        end
+
+        context "when the token is invalid" do
+          before { @response = delete "/api/v1/tasks/#{task.id}/task_items/#{task_item.id}", nil, empty_header }
+
+          it "should return a HTTP 401 error" do
+            expect(@response.status).to eq(401)
+          end
+
+          it "should not delete the task" do
```

Comment:
Prefer single-quoted strings when you don't need string interpolation or special symbols.